pkg/config/default_windows.go

```diff
@@ -0,0 +1,41 @@
+//go:build windows && !brew && !archivematica && !js
+// +build windows,!brew,!archivematica,!js
+
+// Copyright 2014 Richard Lehane. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"os"
+	"path/filepath"
+)
+
+
+func userDataDir(home string) string {
+	if path, _ := windows.KnownFolderPath(*windows.FOLDERID_LocalAppData, windows.KF_FLAG_DEFAULT|KF_FLAG_DONT_VERIFY); path == "" {
+		path, _ := windows.KnownFolderPath(*windows.FOLDERID_LocalAppData, windows.KF_FLAG_DEFAULT_PATH|KF_FLAG_DONT_VERIFY)
+	}
+
+	if path == "" {
```

Comment:
Golang on Windows is suggesting that path may not be initialized by here. I'm not sure why the conditional above isn't doing that correctly?

Reply:
Hm, not sure, was this with the missing import already added back in?

Reply:
While I have the benefit of a Windows machine, I think this (should?) give you what you want here:

```diff
diff --git a/pkg/config/default_windows.go b/pkg/config/default_windows.go
index 1abc346f..1bd73611 100644
--- a/pkg/config/default_windows.go
+++ b/pkg/config/default_windows.go
@@ -26,8 +26,9 @@ import (


 func userDataDir(home string) string {
-       if path, _ := windows.KnownFolderPath(*windows.FOLDERID_LocalAppData, windows.KF_FLAG_DEFAULT|KF_FLAG_DONT_VERIFY); path == "" {
-               path, _ := windows.KnownFolderPath(*windows.FOLDERID_LocalAppData, windows.KF_FLAG_DEFAULT_PATH|KF_FLAG_DONT_VERIFY)
+       var path string
+       if path, _ = windows.KnownFolderPath(windows.FOLDERID_LocalAppData, windows.KF_FLAG_DEFAULT|windows.KF_FLAG_DONT_VERIFY); path == "" {
+               path, _ = windows.KnownFolderPath(windows.FOLDERID_LocalAppData, windows.KF_FLAG_DEFAULT_PATH|windows.KF_FLAG_DONT_VERIFY)
        }

        if path == "" {
```

I think you need the dereferenced flag, and then the `KF_FLAG_DONT_VERIFY` needs to module it's from. The path seems to be scoped to the IF at first (ref)[https://www.calhoun.io/one-liner-if-statements-with-errors/#variable-scopes], so perhaps it can be simplified to:

```diff
diff --git a/pkg/config/default_windows.go b/pkg/config/default_windows.go
index 1abc346f..06b6073a 100644
--- a/pkg/config/default_windows.go
+++ b/pkg/config/default_windows.go
@@ -26,10 +26,7 @@ import (


 func userDataDir(home string) string {
-       if path, _ := windows.KnownFolderPath(*windows.FOLDERID_LocalAppData, windows.KF_FLAG_DEFAULT|KF_FLAG_DONT_VERIFY); path == "" {
-               path, _ := windows.KnownFolderPath(*windows.FOLDERID_LocalAppData, windows.KF_FLAG_DEFAULT_PATH|KF_FLAG_DONT_VERIFY)
-       }
-
+       path, _ := windows.KnownFolderPath(windows.FOLDERID_LocalAppData, windows.KF_FLAG_DEFAULT|windows.KF_FLAG_DONT_VERIFY)
        if path == "" {
                dataDir, found := os.LookupEnv("LOCALAPPDATA")
                if found && dataDir != "" {
@@ -38,6 +35,5 @@ func userDataDir(home string) string {
                        path = filepath.Join("AppData", "Local")
                }
        }
-
        return xdgPath(home, path)
 }
```

If i read it correctly?

Reply:
Yeah, initializing the `path` variable outside the if-statement will work, I wasn't paying enough attention to the scoping rules involved. I'll fix it in a follow-up commit soon.

Your latter suggestion drops the check against `windows.KF_FLAG_DEFAULT`. The flags are documented [here](https://learn.microsoft.com/en-us/windows/win32/api/shlobj_core/ne-shlobj_core-known_folder_flag), the logic was adapted from the `xdg` library initially proposed. As far as I understand the intent, this is because it is possible to have the folders redirected (somehow, I'm not enough of a Windows expert and didn't get to look at this more closely), but there is likely no guarantee that the redirect will lead to a working location so a fallback to the default path is used. Do you think this can be simplified or won't work as is?

Reply:
> Do you think this can be simplified or won't work as is?

I was just reading the code wrong with that proposed "simplification" 😊 with the scoping corrected then I'm sure your original intention is the way to go. (although, also speaking as a non-Windows expert!)

Reply:
Should be fixed now.